flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/flip1/PipelinedRegionComputeUtil.java

```diff
@@ -45,7 +45,7 @@
 	private static final Logger LOG = LoggerFactory.getLogger(PipelinedRegionComputeUtil.class);
 
 	public static Set<PipelinedRegion> toPipelinedRegionsSet(
-			final Set<? extends Set<? extends SchedulingExecutionVertex>> distinctRegions) {
+			final Set<? extends Set<SchedulingExecutionVertex>> distinctRegions) {
```

Comment:
It can be further simplified to Set<Set<SchedulingExecutionVertex>>.
The return type of below method `toExecutionVertexIdSet()` can be simplified as well.

Reply:
Here one can apply the PECS principle.

A signature like
```
final Set<? extends Set<? extends SchedulingExecutionVertex>> distinctRegions
```

enables the following additional invocations:

```
			Set<HashSet<SchedulingExecutionVertex>> b = null;
			PipelinedRegionComputeUtil.toPipelinedRegionsSet(b);
			
			Set<Set<DefaultExecutionVertex>> c = null;
			PipelinedRegionComputeUtil.toPipelinedRegionsSet(c);
```

Therefore, I'd evenn say that my previous fixup commit was wrong.

Reply:
You are right. For the method definition, we should keep it as to stay more powerful.
For the usages in RestartPipelinedRegionFailoverStrategy and RegionPartitionReleaseStrategy, we can simplify it.